lib/functions/settings_spec.lua

```diff
@@ -1,30 +1,42 @@
-local settings = import("./settings")
+local SettingsGenerator = import("./settings")
```

Comment:
This is a flag to me that we should probably rename `settings` to something else, like `createSettings`. That would probably make this PR as a whole a bit more clear.